schema/collection.go

```diff
@@ -93,6 +93,10 @@ func (d *DefaultCollection) GetPrimaryKey() *Index {
 	return d.PrimaryKey
 }
 
+func (d *DefaultCollection) GetPrimaryQueryableFields() []*QueryableField {
```

Comment:
I am not seeing the usage of it, is it needed?

Reply:
Ah yes. I was planning to remove the generics in key_builder and use this. I'll remove and add it later